client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/usecase/LoginUserUseCase.kt

```diff
@@ -0,0 +1,46 @@
+package org.thechance.common.domain.usecase
+
+import org.thechance.common.domain.entity.UserTokens
+import org.thechance.common.domain.getway.ILocalGateway
+import org.thechance.common.domain.getway.IRemoteGateway
+
+interface ILoginUserUseCase {
+
+    suspend fun saveAccessToken(token: String)
+
+    suspend fun saveRefreshToken(token: String)
+
+    suspend fun getAccessToken(): String
+
+    suspend fun getRefreshToken(): String
+
+    suspend fun loginUser(username: String, password: String, keepLoggedIn: Boolean): UserTokens
+
+}
+
+class LoginUserUseCase(
+    private val localGateway: ILocalGateway,
+    private val remoteGateway: IRemoteGateway
+) : ILoginUserUseCase {
+
+    override suspend fun saveAccessToken(token: String) {
+        localGateway.saveAccessToken(token)
+    }
+
+    override suspend fun saveRefreshToken(token: String) {
+        localGateway.saveRefreshToken(token)
+    }
+
+    override suspend fun getAccessToken(): String {
+        return localGateway.getAccessToken()
+    }
+
+    override suspend fun getRefreshToken(): String {
+        return localGateway.getRefreshToken()
+    }
+
+    override suspend fun loginUser(username: String, password: String, keepLoggedIn: Boolean): UserTokens {
+        return remoteGateway.loginUser(username, password, keepLoggedIn)
```

Comment:
Why are you passing the local flag to login method?!!

Reply:
It's the usecase, where we can access the local and remote gateways, and It's a rule of our app to save the login preferences